### PR TITLE
feat:Added support for redshift connector

### PR DESCRIPTION
**Pull request by varunmaddipati**
Adding support for a redshift connector. 
Reference : https://docs.aws.amazon.com/redshift/latest/mgmt/welcome.html

| Q                       | A                                |
| ----------------------- | -------------------------------- |
| Fixed issues?           |  |
| Tests added + passed?   | Yes                              |
| Changeset added?        | Yes                              |
| Documentation PR link   |                                  |
| Any dependency changes? | No                               |

### CODE REVIEW

File: packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx [hunk 1]
```diff
@@ -40,13 +40,16 @@ import {
   DelegatedKerberosAuthenticationStrategy,
   OAuthAuthenticationStrategy,
   SnowflakePublicAuthenticationStrategy,
+  TestDatabaseAuthenticationStrategy,
```
Comment (hardikmaheshwari) [inline packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx:43]: Why is this import needed - `TestDatabase...`

File: packages/legend-studio/src/stores/__tests__/roundtrip/RelationalRoundtripTestData.ts [hunk 2]
```diff
@@ -3812,16 +3812,17 @@ export const testRelationalDatabaseConnectionRoundtrip = [
       connectionValue: {
         _type: 'RelationalDatabaseConnection',
         authenticationStrategy: {
-          _type: 'oauth',
-          oauthKey: 'dummy',
-          scopeName: 'UserPass',
+          _type: 'userPassword',
```
Comment (hardikmaheshwari) [inline packages/legend-studio/src/stores/__tests__/roundtrip/RelationalRoundtripTestData.ts:3815]: We should not be modifying existing test
We need to append our test-data

File: packages/legend-studio/src/stores/editor-state/element-editor-state/connection/ConnectionEditorState.ts [hunk 3]
```diff
@@ -178,6 +184,12 @@ export class RelationalDatabaseConnectionValueState extends ConnectionValueState
         );
         return;
       }
+      case CORE_DATASOURCE_SPEC_TYPE.REDSHIFT: {
+        this.connection.setDatasourceSpecification(
+          new RedshiftDatasourceSpecification('', '', 5439),
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/editor-state/element-editor-state/connection/ConnectionEditorState.ts:189]: can we default to port 80 to make it consistent with the other default port numbers, or is there a reason we default to 5439?

Comment (varunmaddipati) [reply packages/legend-studio/src/stores/editor-state/element-editor-state/connection/ConnectionEditorState.ts:189]: I used 5439 because that is the port that Redshift uses by default

**Review by MauricioUyaguari — APPROVED** [review]
